src/passes/strings-encoding/StringEncoding.cpp

```diff
@@ -487,20 +486,101 @@ bool StringEncoding::processGlobal(BasicBlock& BB, Instruction&, Use& Op, Global
   return true;
 }
 
+static Expected<std::string> runClangExecutable(StringRef code, StringRef dashx,
+                                                StringRef triple,
+                                                std::vector<StringRef> args) {
+  std::vector<StringRef> cmd;
+  cmd.reserve(args.size() + 6);
+
+  static int Dummy;
+  std::string clang = sys::fs::getMainExecutable("clang", (void *)&Dummy);
+  cmd.push_back(clang);
+  cmd.push_back("-S");
+  cmd.push_back("-emit-llvm");
+
+  for (StringRef arg : args)
```

Comment:
Unnecessary copy.
```suggestion
  for (const auto &arg : args)
    cmd.push_back(arg);
```

Reply:
That's idiomatic usage. `StringRef` has value semantics.

Reply:
Totally true!